fields.go

```diff
@@ -16,6 +16,9 @@ type Field = zapcore.Field
 var (
 	// String constructs a field with the given key and value.
 	String = zap.String
+	// String constructs a field with the given key and value. The returned field will
+	// safely and explictly represent `nil` when appropriate.
+	StringPtr = zap.Stringp
```

Comment:
Should we keep the same name?

Reply:
I'm hesitating on this, I feel that `Stringp` is really easy to miss. Given that this isn't something we're going to use too often easier, I gave it a try with `Ptr`. 

No strong opinion there, there's value in keeping the same naming as `zap`, but we shouldn't overplay it.

Reply:
Just adding my vote here. I would prefer `Ptr` too and it's also consistent with the rest of the codebase based on a search I did https://sourcegraph.com/search?q=context:global+repo:sourcegraph/sourcegraph+Ptr%7CStringp&patternType=regexp

The search might be confirmation bias though 😆